src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentSynchronization/DocumentDidChangeEndpoint.cs

```diff
@@ -72,8 +73,18 @@ internal SourceText ApplyContentChanges(IEnumerable<TextDocumentContentChangeEve
             }
 
             var startLinePosition = new LinePosition(change.Range.Start.Line, change.Range.Start.Character);
+            if (!sourceText.IsLinePositionValid(startLinePosition, _logger))
+            {
+                continue;
+            }
+
             var startPosition = sourceText.Lines.GetPosition(startLinePosition);
             var endLinePosition = new LinePosition(change.Range.End.Line, change.Range.End.Character);
+            if (!sourceText.IsLinePositionValid(endLinePosition, _logger))
+            {
+                continue;
+            }
+
             var endPosition = sourceText.Lines.GetPosition(endLinePosition);
```

Comment:
As above, `startPosition` and `endPosition` could both be calculated by calling `change.Range.{Start/End}.TryGetAbsoluteIndex(...)`

Reply:
Agreed, with the same comment - seems like it would still be good to log error here and record fault telemetry. Thoughts?

Reply:
Fixed, thanks.